src/common/json_utils.py

```
@@ -0,0 +1,36 @@
+
+    Raises:
+        TypeError: If object type is not JSON serializable
+    """
+    if isinstance(obj, Decimal):
+        # Round to 2dp for JSON output
+        return float(round(obj, 2))
+    elif isinstance(obj, datetime):
+        return obj.isoformat()
+    elif hasattr(obj, "__str__"):
+        return str(obj)
+    raise TypeError(f"Object of type {type(obj).__name__} is not JSON serializable")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)